slides/aml-25-042518-time-series/index.html

```diff
@@ -0,0 +1,952 @@
+lr_poly = make_pipeline(PolynomialFeatures(include_bias=False), 
+                          LinearRegression())
+lr_poly.fit(X_train, train)
+plt.plot(ppm.index, lr_poly.predict(X))
+ppm.plot()
+```
+]
+
+.center[![:scale 55%](images/quadratic model.png)]
+
+
+???
+
+---
+
+# Detrending with trend model
+
+.smaller[
+```python
+y_res = ppm - lr_poly.predict(X)
+y_res.plot()
+```]
+
+.center[![:scale 60%](images/detrending with trend model.png)]
+
+???
+
+---
+class: compact
+
+# AR model on residuals
+
+.left-column[
+.smaller[
+```python
+from statsmodels.tsa import ar_model
+ar_model_res = ar_model.AR(y_res[:500])
+res_res = ar_model_res.fit(maxlag=12)
+ar_pred_res = res_res.predict(
+        ppm.index[500], ppm.index[-1])
+pred_ar_res = ar_pred_res + 
+              r_poly.predict(X_test)
+res_res.params
+```
+```
+const              -0.008039
+L1.interpolated     0.922508
+L2.interpolated    -0.243015
+L3.interpolated    -0.101173
+L4.interpolated    -0.077538
+L5.interpolated     0.158073
+L6.interpolated    -0.006164
+L7.interpolated    -0.023890
+L8.interpolated    -0.126381
+L9.interpolated     0.092664
+L10.interpolated   -0.025640
+L11.interpolated    0.414180
+L12.interpolated   -0.018078
+dtype: float64
+```
+]
+]
+
+.right-column[
+.center[![:scale 120%](images/ar model residuals.png)]
+
+]
+
+
+???
+FIXME - Note: Couldn't decrease size further
+
+
+---
+
+# Just Linear Regression
+
+.smaller[
+```python
+X_month = pd.concat([X, pd.DataFrame({'month': X.index.month}, index=X.index)], axis=1)
+X_month.head()
+```
+]
+.center[![:scale 30%](images/linear regression.png)]
+.smaller[
+```python
+X_train_month = X_month[:500]
+X_test_month = X_month[500:]
+lr_poly_month = make_pipeline(PolynomialFeatures(include_bias=False), LinearRegression())
+lr_poly_month.fit(X_train_month, train)
+X_test_month.shape # (209,2)
+```
+
+]
+
+???
+
+---
+
+.center[![:scale 90%](images/arma month.png)]
+
+???
+
+---
+
+.smaller[
+```python
+from sklearn.preprocessing import OneHotEncoder
+lr_poly_month_ohe = make_pipeline(OneHotEncoder(categorical_features=[1], sparse=False),
+                                  PolynomialFeatures(include_bias=False), LinearRegression())
+lr_poly_month_ohe.fit(X_train_month, train)
+```
+.center[![:scale 55%](images/arma res.png)]
+```python
+from sklearn.metrics import mean_squared_error
+mean_squared_error(ppm[500:], lr_poly_month_ohe.predict(X_test_month)) #0.505
+mean_squared_error(ppm[500:], pred_arma_res) #0.577
+```
+]
+
+???
+
+---
+class: spacious
+
+# General Autoregressors
+
+- https://github.com/scikit-learn/scikit-learn/pull/6029
+- Trees work well (as always)
+- Can take explanatory variables
+
+???
+
+---
+
+# Multivariate Timeseries
+
+.smaller[
+- energy data from https://archive.ics.uci.edu/ml/datasets/Appliances+energy+prediction
+
+```python
+energy = pd.read_csv("energydata_complete.csv", parse_dates=['date'], 
+                      index_col="date")
```

Comment:
indentation

Reply:
didn't get the error

Reply:
``index_col`` is indented one space too much. I can fix it.